### PR TITLE
Sync product images

### PR DESCRIPTION
**Pull request by trickreich**

### CODE REVIEW

File: Model/Product/Handler/Message/SynchronizeProductMessageHandler.php [hunk 1]
```diff
@@ -103,4 +132,51 @@ protected function synchronizeTranslation(
...
+
+            $addedMediaReference->setSorting($sorting);
+            $sorting++;
+        }
+    }
+
+    protected function synchronizeImage(ProductImageValueObject $imageValueObject, ProductInterface $product): ?ProductMediaReference {
+        // load image
+        try {
+            $fileContents = file_get_contents('http://sylius.localhost/media/image/' . $imageValueObject->getPath());
```
Comment (trickreich) [inline Model/Product/Handler/Message/SynchronizeProductMessageHandler.php]: config value

File: Model/Product/Handler/Message/SynchronizeProductMessageHandler.php [hunk 2]
```diff
@@ -103,4 +146,102 @@ protected function synchronizeTranslation(
         $productInformation->setMetaDescription($translationValueObject->getMetaDescription());
         $productInformation->setShortDescription($translationValueObject->getShortDescription());
     }
+
+    protected function synchronizeImages(SynchronizeProductMessage $message, ProductInterface $product): void
+    {
+        $sorting = 1;
+        foreach ($message->getImages() as $imageValueObject) {
+            $mediaReference = $this->productMediaReferenceRepository->findBySyliusId($imageValueObject->getId());
```
Comment (wachterjohannes) [inline Model/Product/Handler/Message/SynchronizeProductMessageHandler.php]: extract this to an own function: 

```
$mediaReference = $this->productMediaReferenceRepository->findBySyliusId($imageValueObject->getId());

if (!$mediaReference) {
    $this->createMediaReference($imageValueObject, $product, $sorting);

    return;
}

if ($mediaReference->getSyliusPath() !== $imageValueObject->getPath()) {
    $this->updateMediaReference($imageValueObject, $product, $mediaReference, $sorting);

    return;
}

$this->mediaFactory->updateTitles($mediaReference->getMedia(), $this->getImageTitles($product));
```

would make this more readable

File: Model/Media/MediaFactory.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SyliusConsumerBundle\Model\Media;
```
Comment (wachterjohannes) [inline Model/Media/MediaFactory.php]: would move this to a namepsace outside of model

File: Model/Media/MediaFactory.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+use Sulu\Bundle\MediaBundle\Entity\MediaInterface;
+use Sulu\Bundle\MediaBundle\Entity\MediaRepository;
+use Sulu\Bundle\MediaBundle\Entity\MediaType;
+use Sulu\Bundle\MediaBundle\Media\Storage\StorageInterface;
+use Sulu\Bundle\MediaBundle\Media\TypeManager\TypeManager;
+use Sulu\Component\Media\SystemCollections\SystemCollectionManagerInterface;
+use Symfony\Component\HttpFoundation\File\File as SymfonyFile;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class MediaFactory
```
Comment (wachterjohannes) [inline Model/Media/MediaFactory.php]: interface

File: Resources/config/services.xml [hunk 5]
```diff
@@ -38,5 +38,14 @@
...
         </service>
+
+        <service id="Sulu\Bundle\SyliusConsumerBundle\Model\Media\MediaFactory">
+            <argument type="service" id="sulu_media.storage"/>
+            <argument type="service" id="sulu.repository.media"/>
+            <argument type="service" id="sulu_media.type_manager"/>
+            <argument type="service" id="doctrine.orm.entity_manager"/>
+            <argument type="service" id="sulu_media.system_collections.manager"/>
+        </service>
+
```
Comment (wachterjohannes) [inline Resources/config/services.xml]: remove empty line

**Pull request merged**